test/unit-test/FreeRTOS_DHCP/FreeRTOS_DHCP_utest.c

```diff
@@ -0,0 +1,3657 @@
+        /* Ignore the inputs to setting the socket options. */
+        FreeRTOS_setsockopt_IgnoreAndReturn( pdPASS );
+        /* Make sure that binding fails. Return anything except zero. */
+        vSocketBind_IgnoreAndReturn( pdTRUE );
+        /* Then expect the socket to be closed. */
+        vSocketClose_ExpectAndReturn( &xTestSocket, NULL );
+        /* See if the timer is reloaded. */
+        vIPReloadDHCPTimer_Expect( dhcpINITIAL_TIMER_PERIOD );
+        /* Try all kinds of states. */
+        vDHCPProcess( pdTRUE, i );
+
+        /* Expected state is incorrect, but we are trying to reset
+         * the DHCP the state machine. */
+        TEST_ASSERT_EQUAL( eWaitingSendFirstDiscover, xDHCPData.eDHCPState );
+        TEST_ASSERT_EQUAL( xDHCPSocket, NULL );
+        TEST_ASSERT_EQUAL( xDHCPData.xUseBroadcast, 0 );
+        /* This should be reset as well */
+        TEST_ASSERT_EQUAL( xDHCPData.ulOfferedIPAddress, 0 );
+        /* And this too. */
+        TEST_ASSERT_EQUAL( xDHCPData.ulDHCPServerAddress, 0 );
+    }
+}
+
+void test_vDHCPProcess_ResetAndIncorrectStateWithRNGSuccessSocketSuccess( void )
+{
+    struct xSOCKET xTestSocket;
+
+    /* Test all the valid and invalid entries. */
+    for( int i = 0; i < ( eNotUsingLeasedAddress * 2 ); i++ )
+    {
+        /* This should get assigned to a given value. */
+        xDHCPSocket = NULL;
+        /* Put any state. */
+        xDHCPData.eDHCPState = eSendDHCPRequest;
+        /* This should be reset to 0. */
+        xDHCPData.xUseBroadcast = 1;
+        /* This should be reset as well */
+        xDHCPData.ulOfferedIPAddress = 0xAAAAAAAA;
+        /* And this too. */
+        xDHCPData.ulDHCPServerAddress = 0xABABABAB;
+
+
+        /* Make random number generation pass. */
+        xApplicationGetRandomNumber_ExpectAndReturn( &( xDHCPData.ulTransactionId ), pdTRUE );
+        /* Return a valid socket. */
+        FreeRTOS_socket_ExpectAndReturn( FREERTOS_AF_INET, FREERTOS_SOCK_DGRAM, FREERTOS_IPPROTO_UDP, &xTestSocket );
+        /* Ignore the inputs to setting the socket options. */
+        FreeRTOS_setsockopt_IgnoreAndReturn( pdPASS );
+        /* Make sure that binding fails. Return anything except zero. */
+        vSocketBind_IgnoreAndReturn( 0 );
+        /* See if the timer is reloaded. */
+        vIPReloadDHCPTimer_Expect( dhcpINITIAL_TIMER_PERIOD );
+        /* Try all kinds of states. */
+        vDHCPProcess( pdTRUE, i );
+
+        /* Expected state is incorrect, but we are trying to reset
+         * the DHCP the state machine. */
+        TEST_ASSERT_EQUAL( eWaitingSendFirstDiscover, xDHCPData.eDHCPState );
+        TEST_ASSERT_EQUAL( xDHCPSocket, &xTestSocket );
+        TEST_ASSERT_EQUAL( xDHCPData.xUseBroadcast, 0 );
+        /* This should be reset as well */
+        TEST_ASSERT_EQUAL( xDHCPData.ulOfferedIPAddress, 0 );
+        /* And this too. */
```

Comment:
should the the other way arround

TEST_ASSERT_XXX ( expected_value, variable_to_test ); as doing it the other way around makes the message output by unity on failure look weird 

there are a few other occurrences of this in this file.